### PR TITLE
Self contain blueprint boshrelease

### PR DESCRIPTION
**Pull request by akskap**
https://jtrack.wdf.sap.corp/browse/NGPBUG-38447

### CODE REVIEW

File: packages/blueprint/packaging [hunk 1]
```diff
@@ -16,21 +19,40 @@ echo "Rebuild NPM dependencies..."
...
+fi
+
+echo "Installing pip..."
+echo "Creating the pip site packages..."
+mkdir -p /var/vcap/packages/python/lib/python3.4/site-packages
+
+echo "Setting the PYTHONPATH with setuptools and pip site packages..."
+export PYTHONPATH=/var/vcap/packages/python/lib/python3.4/site-packages:$PYTHONLIBS
+
+pushd pip-${PIP_VERSION}
```
Comment (pritishmishra) [inline packages/blueprint/packaging]: Will it help to add a comment on what this does? I mean .. I had to google it. So, just asking! :)

File: packages/blueprint/packaging [hunk 2]
```diff
@@ -16,21 +19,40 @@ echo "Rebuild NPM dependencies..."
...
+else
+  echo "Successfully extracted pip-${PIP_VERSION}..."
+fi
+
+echo "Installing pip..."
+echo "Creating the pip site packages..."
+mkdir -p /var/vcap/packages/python/lib/python3.4/site-packages
+
+echo "Setting the PYTHONPATH with setuptools and pip site packages..."
+export PYTHONPATH=/var/vcap/packages/python/lib/python3.4/site-packages:$PYTHONLIBS
```
Comment (ketakig12) [inline packages/blueprint/packaging]: Is the intention to override [this](https://github.com/cloudfoundry-incubator/service-fabrik-blueprint-boshrelease/blob/9668e2e035fafb174e74196d3796a08278e5bfb7/jobs/broker-agent/templates/bin/job_properties.sh.erb#L22) export?

File: packages/blueprint/packaging (hunk 1 above)
Comment (akskap) [reply packages/blueprint/packaging]: Sure, added.

File: config/blobs.yml [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 node/node-v6.9.1-linux-x64.tar.xz:
+  size: 9351072
   object_id: c76f3e2b-5864-49b7-9dd0-66bb2526b9e6
   sha: 84b0b6fde43a4d892186119ba6b9f1db3c2c6129
-  size: 9351072
+python/Python-3.4.3.tar.xz:
+  size: 14421964
+  object_id: e4a90d84-f05e-4a8d-8d59-a7f26809afdb
+  sha: 7ca5cd664598bea96eec105aa6453223bb6b4456
+python/pip-18.1.tar.gz:
```
Comment (jayeshjk) [inline config/blobs.yml]: Does this mean that we need to manually take care of pip updates from now on? Asking this from the BnR library perspective wherein we might upgrade SDKs of various IAAS which might require latest versions of pip as well.

File: packages/blueprint/packaging [hunk 4]
```diff
@@ -16,21 +19,41 @@ echo "Rebuild NPM dependencies..."
...
+pip3 install --target=${PYTHONLIBS} * --no-index --find-links=file://${SOURCES_PATH} --ignore-installed
 # Remove python2 backport module 'futures'
 # In python 3, the functionalities of 'futures' are builtin. As the python-swiftclient package is compatiable for both
 #   python2 and python3, it lists 'futures' in its dependencies list. We have to delete it to prevent python from using
 #   the functionalities of this module which are incompatible with python3 itself.
 # Unfortunately, pip has no mechanism to uninstall a package from a specific directory. Thus, we have to delete it manually.
-rm -rf ${PYTHONPATH}/concurrent
-rm -rf ${PYTHONPATH}/futures*
+rm -rf ${PYTHONLIBS}/concurrent
+rm -rf ${PYTHONLIBS}/futures*
```
Comment (ketakig12) [inline packages/blueprint/packaging]: Is [this](https://github.com/cloudfoundry-incubator/service-fabrik-blueprint-boshrelease/blob/9668e2e035fafb174e74196d3796a08278e5bfb7/packages/blueprint/pre_packaging#L17-L18) used anytime? I guess not, but if used, also need to be changed / removed.

File: config/blobs.yml (hunk 3 above)
Comment (akskap) [reply config/blobs.yml]: Yes, intention behind this change is to a) self contain and b) control the pip version being installed.

File: packages/blueprint/packaging (hunk 2 above)
Comment (akskap) [reply packages/blueprint/packaging]: No, that is not intended here. I doubt if that ENV variable is actually available during execution of this script

File: packages/blueprint/packaging (hunk 4 above)
Comment (akskap) [reply packages/blueprint/packaging]: Those commands are referred in 'pre_packaging' hence should not be a problem here

**Pull request merged**